### PR TITLE
[DW-0000] Add home page layout and supporting components

### PR DESCRIPTION
**Pull request by robValtech**
New components:
- Search banner
- Publication list
- Services
- Document picker

Misc updates:
- Create new abstract page for the homeland page
- Make the grid layout fully responsive

### CODE REVIEW

File: repository-data/webfiles/src/main/resources/site/freemarker/common/document-list/services-list.ftl [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<#ftl output_format="HTML">
+<#include "../../include/imports.ftl">
+
+<@hst.setBundle basename="homepage.website.labels"/>
+<@fmt.message key="systems-and-services.title" var="title"/>
+<@fmt.message key="systems-and-services.text" var="text"/>
+<@fmt.message key="systems-and-services.subTitle" var="subTitle"/>
+<@fmt.message key="systems-and-services.buttonLabel" var="buttonLabel"/>
```
Comment (wojtek-oledzki) [inline repository-data/webfiles/src/main/resources/site/freemarker/common/document-list/services-list.ftl]: I like that approach - I think we should start using the same approach and get all labels/headers at the top of the template. What do you think @ngta1, @jacsamell @ziemsky

File: repository-data/webfiles/src/main/resources/site/freemarker/common/publications-list.ftl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                    <#if hstRequestContext.preview>
+                    <div style="position:relative">
+                        <@hst.cmseditlink hippobean=item />
+                    </div>
+                    </#if>
+                    <h2><a href="<@hst.link hippobean=item/>">${item.title}</a></h2>
+                    <#assign nomimaldate = item.getNominalPublicationDate()/>
+                    <#if !nomimaldate.isRestricted()>
+                    <#assign pubdate = nomimaldate.dayOfMonth + " " + nomimaldate.month +" "+ nomimaldate.year?c/>
+                        <time datetime="${pubdate?date?iso_utc}T08:00">${pubdate?date}</time>
```
Comment (wojtek-oledzki) [inline repository-data/webfiles/src/main/resources/site/freemarker/common/publications-list.ftl]: "8:00"?

File: repository-data/webfiles/src/main/resources/site/freemarker/common/search-banner.ftl [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<#ftl output_format="HTML">
+<#include "../include/imports.ftl">
+
+<@hst.setBundle basename="homepage.website.labels"/>
+<@fmt.message key="search-banner.placeholder" var="placeholderText"/>
+<@fmt.message key="search-banner.buttonLabel" var="buttonLabel"/>
+<@fmt.message key="search-banner.title" var="searchTitle"/>
+<@fmt.message key="search-banner.text" var="searchBannerText"/>
+
+<#if facets??>
```
Comment (wojtek-oledzki) [inline repository-data/webfiles/src/main/resources/site/freemarker/common/search-banner.ftl]: this is already in `../include/imports.ftl`

File: repository-data/webfiles/src/main/resources/site/freemarker/common/search-banner/search-alternative.ftl [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<#ftl output_format="HTML">
+<#include "../include/imports.ftl">
+
+<@hst.setBundle basename="website.labels"/>
+<@fmt.message key="labels.search-placeholder" var="placeholderText"/>
+<@fmt.message key="labels.search-button" var="buttonText"/>
+<@fmt.message key="labels.search-title" var="searchTitle"/>
+<@fmt.message key="labels.search-banner-text" var="searchBannerText"/>
+
+<#if facets??>
```
Comment (wojtek-oledzki) [inline repository-data/webfiles/src/main/resources/site/freemarker/common/search-banner/search-alternative.ftl]: already in first "include"

File: migrator/src/main/java/uk/nhs/digital/ps/migrator/model/hippo/MappedFields.java [hunk 5]
```diff
@@ -0,0 +1,186 @@
+package uk.nhs.digital.ps.migrator.model.hippo;
```
Comment (wojtek-oledzki) [inline migrator/src/main/java/uk/nhs/digital/ps/migrator/model/hippo/MappedFields.java]: I'm not sure how this file is related to your PR :(

File: repository-data/webfiles/src/main/resources/site/freemarker/common/document-list/services-list.ftl (hunk 1 above)
Comment (jacsamell) [reply repository-data/webfiles/src/main/resources/site/freemarker/common/document-list/services-list.ftl]: Hmmmm not sure, personally I like not having the extra variable but inserting directly into the element instead. If you have the variable `title` for example it's easy to get mixed up with `document.title`. It's also harder to tell from the template what the thing is referring to. If I see a variable in the FTL template I'm probably going to look in the component.java for the variable creation of `title`. I suppose we could name the variables `title-label` but personally I think you may as well insert directly.

I don't mind it being done the way it has, this is just my preference 😁

File: repository-data/application/src/main/resources/hcm-content/content/documents/administration/website.yaml [hunk 6]
```diff
@@ -1,6 +1,151 @@
 ---
 /content/documents/administration/website:
   .meta:order-before: national-indicator-library
+  /homepage-labels:
```
[the remark this answers is not in the document]
Comment (giacomolm) [reply repository-data/application/src/main/resources/hcm-content/content/documents/administration/website.yaml]: @wojtek-oledzki every new deploy is generally executed without content: this means everything you have in the content folder will stay at it is. If you want to bootstrap new content as well, then there are several ways to do it, but you need to take into account the existing content. 

In this case you could use a specific bootstrap module containing the incremental content update.

File: repository-data/webfiles/src/main/resources/site/freemarker/common/publications-list.ftl (hunk 2 above)
Comment (giacomolm) [reply repository-data/webfiles/src/main/resources/site/freemarker/common/publications-list.ftl]: The nominal date returned by getNominalPublicationDate() doesn't contain time, for this reason we made this assumption. I am not sure what the datetime attribute does, maybe I can ask @robValtech if there is a way to specify only the date

File: README.md [hunk 7]
```diff
@@ -137,6 +137,54 @@ There are 2 ways to run the server: with or without `autoexport`. When the app r
 
 In order to be able to test your localhost from a VM (such as VirtualBox) you'll have to add your IPv4 address to the server configuration. You can read how it can be done [here](docs/what-if/use-custom-host.md)
 
+## Valtech branching strategy
```
Comment (jacsamell) [inline README.md]: Might be worthy of it's own readme file / what if file, just so the top level read me can be kept to a clean minimum?

File: repository-data/webfiles/src/main/resources/site/freemarker/common/document-list.ftl [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<#ftl output_format="HTML">
+<#include "../../include/imports.ftl">
+
+<#if pageable?? && pageable.items?has_content>
+  <ul>
```
[the remark this answers is not in the document]
Comment (robValtech) [reply repository-data/webfiles/src/main/resources/site/freemarker/common/document-list.ftl]: @wojtek-oledzki Thanks for the comment, I absolutely agree with you - all our SCSS from the NHS UI kit uses 2 spaces by default. I have been converting them to 4 as I am touching the files and create new ones. Soon we'll only have SCSS files with 4 spaces.

File: repository-data/webfiles/src/main/resources/site/freemarker/common/publications-list.ftl (hunk 2 above)
Comment (robValtech) [reply repository-data/webfiles/src/main/resources/site/freemarker/common/publications-list.ftl]: @wojtek-oledzki Thanks for this, time is not mandatory for a valid `datetime` attribute, so we'll drop it from the code.

File: repository-data/webfiles/src/main/resources/site/freemarker/common/search-banner.ftl (hunk 3 above)
Comment (robValtech) [reply repository-data/webfiles/src/main/resources/site/freemarker/common/search-banner.ftl]: @wojtek-oledzki Thanks for the heads up, I'll remove it.

File: repository-data/webfiles/src/main/resources/site/freemarker/common/search-banner/search-alternative.ftl (hunk 4 above)
Comment (robValtech) [reply repository-data/webfiles/src/main/resources/site/freemarker/common/search-banner/search-alternative.ftl]: @wojtek-oledzki Thanks for the heads up, I'll remove it.

File: migrator/src/main/java/uk/nhs/digital/ps/migrator/model/hippo/MappedFields.java (hunk 5 above)
Comment (robValtech) [reply migrator/src/main/java/uk/nhs/digital/ps/migrator/model/hippo/MappedFields.java]: @wojtek-oledzki Good catch - this isn't supposed to be in, I'll be removing it altogether

File: migrator/src/main/java/uk/nhs/digital/ps/migrator/model/hippo/MappedFields.java [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import uk.nhs.digital.ps.migrator.report.MigrationReport;
+
+import java.util.*;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public final class MappedFields {
```
[the remark this answers is not in the document]
Comment (robValtech) [reply migrator/src/main/java/uk/nhs/digital/ps/migrator/model/hippo/MappedFields.java]: @jacsamell Good catch - this isn't supposed to be in, I'll be removing it altogether

File: README.md (hunk 7 above)
Comment (robValtech) [reply README.md]: Hi @jacsamell This little section is only there to give the developer a reference to an external file. We always add only short sections of important information to the README, and move the lengthy stuff to 'what-ifs'

Comment (jacsamell) [reply README.md]: I meant the section below really Steps 1,2,3,4,5 maybe that could be moved to an external document. I'm not too fussed on this, people can just ignore that section if they want 🙂

**Pull request merged**